### PR TITLE
Edit notes

### LINKED ISSUE #118: [BUG] /del on first line of add-n content
See snippets for details on the issue.

Screenshot_1

### PR DESCRIPTION
**Pull request by Nazryl**
#18 
Fixed #118
Edit notes are now available.

Commands eg: /i /t /add /ln /del /tag /c
edit-n /i 1 /add 2 Line 2 /t Java OOP /tag CS2113 green /tag Hard blue /c
edit-n /i 1 /ln 2 Line 2.5
edit-n /i 1 /del 2 

/add, /ln and /del cannot be used at the same time

### CODE REVIEW

File: src/main/java/seedu/duke/command/EditNoteCommand.java [hunk 1]
```diff
@@ -14,24 +24,109 @@
     public static final String COMMAND_USAGE = COMMAND_WORD + ": Edits a note in the notebook. Parameters: "
             + PREFIX_DELIMITER + PREFIX_INDEX + " INDEX "
             + "[" + PREFIX_DELIMITER + PREFIX_TITLE + " TITLE] "
-            + "[" + PREFIX_DELIMITER + PREFIX_LINE + " LINE]"
+            + "[" + PREFIX_DELIMITER + PREFIX_LINE + " INDEX STRING]"
```
Comment (R-Ramana) [inline src/main/java/seedu/duke/command/EditNoteCommand.java:27]: `INDEX STRING` is not very clear on what it does. Do you think we can use something like Line number or something?

File: src/main/java/seedu/duke/util/Parser.java [hunk 2]
```diff
@@ -172,7 +179,7 @@ public Command parseCommand(String userInput) {
      * Checks if an input string if blank. If it is, throw the provided system exception. If it is not, return that
      * string trimmed.
      *
-     * @param input Input to be checked.
+     * @param input         Input to be checked.
```
Comment (R-Ramana) [inline src/main/java/seedu/duke/util/Parser.java:182]: As mentioned previously, I don't think the other JavaDoc comments had white space. Maybe we should standardize?

File: src/main/java/seedu/duke/data/exception/SystemException.java [hunk 3]
```diff
@@ -35,15 +35,21 @@
         EXCEPTION_MISSING_INDEX_PREFIX("Missing index prefix!"),
         EXCEPTION_MISSING_INDEX("Missing index!"),
 
+        EXCEPTION_MISSING_LINE("Missing line content!"),
```
Comment (R-Ramana) [inline src/main/java/seedu/duke/data/exception/SystemException.java:38]: Not 100% sure if a user will understand what exactly Missing line content is. maybe can phrase it better?

File: src/main/java/seedu/duke/util/Parser.java [hunk 4]
```diff
@@ -855,7 +1007,7 @@ private Command preparePin(String userMessage) throws SystemException {
      * Returns a CreateTagCommand or a DeleteTagCommand based on the user message.
      *
      * @param userMessage Original string of the user message.
-     * @param isCreate determines the type of command (CreateTagCommand or DeleteTagCommand) to return.
+     * @param isCreate    determines the type of command (CreateTagCommand or DeleteTagCommand) to return.
```
Comment (R-Ramana) [inline src/main/java/seedu/duke/util/Parser.java:1010]: As above for the whitespace

File: src/main/java/seedu/duke/util/Parser.java [hunk 5]
```diff
@@ -941,7 +1093,7 @@ private Command prepareTag(String userMessage) throws SystemException {
      * Sets up what to be printed to the console (only logs that are severe)
      * Sets up what to be printed to the file (logs that are of Level.INFO and above)
      *
-     * @param logger A Logger variable to be used
+     * @param logger      A Logger variable to be used
```
Comment (R-Ramana) [inline src/main/java/seedu/duke/util/Parser.java:1096]: As above for the whitespace

**Review by R-Ramana — APPROVED** [review]

File: src/main/java/seedu/duke/util/Parser.java [hunk 6]
```diff
@@ -404,9 +414,12 @@ private Command prepareAddEvent(String userMessage) throws SystemException {
             Scanner input = new Scanner(System.in);
             inputString = new ArrayList<>();
 
-            System.out.println("Enter Note:");
-            System.out.println("Type /del to delete your previous line");
-            System.out.println("Type /end on a new line to end your note input");
+            String[] instructions = new String[3];
+            instructions[0] = "Enter Note:";
+            instructions[1] = "*/del to delete previous line*";
+            instructions[2] = "*/end on a new line to end note input*";
```
Comment (Chongjx) [inline src/main/java/seedu/duke/util/Parser.java:420]: Can declare this as a const?

**Review by Chongjx — APPROVED** [review]
Looks good to me overall. On a side note, can you add the case for PREFIX_ARCHIVE in the prepare function for adding note?

File: src/main/java/seedu/duke/util/Parser.java (hunk 2 above)
Comment (Nazryl) [reply src/main/java/seedu/duke/util/Parser.java:182]: Im not the author for this function. Ill make the changes.

File: src/main/java/seedu/duke/data/exception/SystemException.java (hunk 3 above)
Comment (Nazryl) [reply src/main/java/seedu/duke/data/exception/SystemException.java:38]: it just means that the contents for that line is missing

File: src/main/java/seedu/duke/util/Parser.java (hunk 2 above)
Comment (Nazryl) [reply src/main/java/seedu/duke/util/Parser.java:182]: I know why already. the ide by default turns on whose whitespaces when auto format

File: src/main/java/seedu/duke/command/EditNoteCommand.java (hunk 1 above)
Comment (Nazryl) [reply src/main/java/seedu/duke/command/EditNoteCommand.java:27]: its not a line number. fixed it

**Comment by Nazryl** [comment]
Settled on adding PREFIX_ARCHIVE case

**Pull request merged**